internal/generator/generator_util.go

```diff
@@ -0,0 +1,238 @@
+/**
+ * Copyright (c) F5, Inc.
+ *
+ * This source code is licensed under the Apache License, Version 2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+package generator
+
+import (
+	_ "embed"
+	"errors"
+	"fmt"
+	"html/template"
+	"io"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+)
+
+// regex.
+const (
+	// Extract single directive definition block
+	// static ngx_command_t  {name}[] = {definition}
+	// this regex extracts {name} and {definition}.
+	extractDirectivesDefBlock = "ngx_command_t\\s+(\\w+)\\[\\]\\s*=\\s*{(.*?)};"
+
+	// Extract one directive definition and attributes from extracted block
+	// { ngx_string({directive_name}),
+	//   {bitmask1|bitmask2|...},
+	//   ... },
+	// this regex extracts {directive_name} and {bitmask1|bitmask2|...}.
+	extractSingleDirective = "ngx_string\\(\"(.*?)\"\\).*?,(.*?),"
```

Comment:
Can you explain what these regex patterns do differently from the regex patterns used by @dareste in his similar tool?

```
ngx_command_t(.*?)\[](?s)(.*?)ngx_null_command
ngx_string\("(.*?)"\),(.*?),
```

You can use backticks to avoid needing to escape `"` characters.

Reply:
> Can you explain what these regex patterns do differently from the regex patterns used by @dareste in his similar tool?

His `ngx_command_t(.*?)\[](?s)(.*?)ngx_null_command` and my `extractDirectivesDefBlock` have the same function. But I'm afraid some authors may not put ngx_null_command at the end of definition(still align with C grammar). So I choose the way of `extractDirectivesDefBlock` (I think "};" must be the end of the definition according to C grammar).

His `ngx_string\("(.*?)"\),(.*?)` and my `extractSingleDirective` are same but I did't use backticks, I will change to backticks